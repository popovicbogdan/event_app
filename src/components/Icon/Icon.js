import React from "react";
import { PropTypes } from "prop-types";
import { IconStyled } from "./Icon.styled";

const Icon = ({ children }) => {
  return <IconStyled>{children}</IconStyled>;
};

Icon.propTypes = {
  children: PropTypes.string
};
export default React.memo(Icon);
