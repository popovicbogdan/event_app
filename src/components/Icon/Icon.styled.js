import styled from "styled-components";

export const IconStyled = styled.i.attrs({
  className: "material-icons",
  "data-testid": "icon"
})`
  && {
    line-height: inherit;
  }
`;
