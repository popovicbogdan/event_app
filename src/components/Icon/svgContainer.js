import React from 'react';
import PropTypes from 'prop-types';

function svgContainer(Icon) {
  function SVG({ width, height, viewBox, color, onClick }) {
    return (
      <svg
        width={width}
        height={height}
        viewBox={viewBox}
        fill={color}
        onClick={onClick}
      >
        <Icon />
      </svg>
    );
  }

  SVG.defaultProps = {
    width: '24',
    height: '24',
    vewBox: '0 0 24 24',
    color: 'black'
  };
  SVG.propTypes = {
    width: PropTypes.string,
    height: PropTypes.string,
    viewBox: PropTypes.string,
    color: PropTypes.string,
    onClick: PropTypes.func
  };
  return SVG;
}

export default svgContainer;
