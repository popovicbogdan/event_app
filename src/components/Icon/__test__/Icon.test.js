import { shallow, configure } from "enzyme";
import Icon from "../Icon";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });
describe("<Icon/>", () => {
  it("should render children correctly", () => {
    const component = shallow(<Icon>search</Icon>);

    expect(component.contains("search")).toBeTruthy();
  });
  it("should match snapshot", () => {
    const component = renderer.create(<Icon>search</Icon>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
