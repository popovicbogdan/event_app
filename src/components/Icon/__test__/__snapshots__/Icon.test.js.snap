// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon/> should match snapshot 1`] = `
<i
  className="sc-AykKC fNAoVs material-icons"
  data-testid="icon"
>
  search
</i>
`;
